torch/linalg/__init__.py

```diff
@@ -248,42 +268,60 @@
 eig = _add_docstr(_linalg.linalg_eig, r"""
 linalg.eig(input, *, out=None) -> (Tensor, Tensor)
 
-Computes the eigenvalues and eigenvectors of a square
-matrix :attr:`input`, or of each such matrix in a batched :attr:`input`.
+Computes the eigenvalue decomposition of a square matrix or batch of square matrices.
```

Comment:
Continuing to experiment with templating, this could be "Computes the eigenvalue decomposition of square matrices."

Reply:
Another option would be to talk exclusively about the batched case and treat single "unbatched" matrices as just having empty batch dimensions. That might be a lot simpler and less conceptually confusing to both writers and readers?

Reply:
In my opinion, it's the other way around. It's much easier to talk about the non-batched case, as the examples you write them for one matrix such as `A = VSV^{-1}`. Writing this for batched inputs is unnecessarily verbose.

I think that we should find patterns that concentrate in conveying the idea of what happens for one matrix. In particular, the we should balance
1) Making clear that everything works for batches
2) Making clear that batches are "just an optimisation", "something convenient", and that what's important to understand is the matrix case.

I really liked the template you proposed above. As such, I'd write this one as:

> Computes the eigenvalues and eigenvectors of a square matrix if they exist. Supports batched and unbatched inputs.

Note how I also slipped in the "if they exist", as the eigenvalues as defined here only exist for diagonalisable matrices. This point is addressed below, but I think it's good to point out that some operations may not be well-defined (e.g. the inverse of a non-invertible matrix). I elaborate on this point in the documentation of this operation.